### PR TITLE
chore(build): Address beta Rust lint warnings

### PR DESCRIPTION
**Pull request by olksdr**
Make sure we can compile Relay with beta toolchain. 

#skip-changelog

### CODE REVIEW

File: relay-event-derive/src/lib.rs [hunk 1]
```diff
@@ -440,6 +440,7 @@ impl FieldAttrs {
 }
 
 #[derive(Copy, Clone, Default)]
+#[allow(dead_code)]
```
Comment (Dav1dde) [inline relay-event-derive/src/lib.rs]: Is this actually still used, Rust Analyzer tells me it's unused

**Review by Dav1dde — APPROVED** [review]

File: relay-event-derive/src/lib.rs (hunk 1 above)
Comment (olksdr) [reply relay-event-derive/src/lib.rs]: That's a question, do we want to keep this around or just delete?

Comment (Dav1dde) [reply relay-event-derive/src/lib.rs]: If it is actually unused, I'd delete it.

Comment (olksdr) [reply relay-event-derive/src/lib.rs]: Makes sense. 
Done in [`57550e6`](https://github.com/getsentry/relay/pull/3092/commits/57550e6fad22a52509c83d2489d00501391acccd)

**Review by iker-barriocanal — APPROVED** [review]
Thank you for doing this!

**Pull request merged**